site/zenodo_rdm/legacy/serializers/schemas.py

```diff
@@ -321,8 +496,16 @@ def dump_state(self, result, original, **kwargs):
     @post_dump(pass_original=True)
     def dump_prereserve_doi(self, result, original, **kwargs):
         """Dump prereserved DOI information."""
+        provider = original["pids"]["doi"]["provider"]
+        recid = original["id"]
+        # For external DOIs, the prereserve_doi is hacked on the response.
+        # TODO confirm this  hack
+        if provider == "external":
+            doi = f"10.5281/{recid}"
```

Comment:
minor: should be `doi = f"10.5281/zenodo.{recid}"` [as in legacy](https://github.com/zenodo/zenodo/blob/b4a4ab88238619811488777f6a2cc92b5d490a94/zenodo/modules/records/minters.py#LL43C26-L43C26).